buildSrc/src/main/java/quilt/internal/tasks/EnigmaProfileConsumingTask.java

```diff
@@ -0,0 +1,49 @@
+package quilt.internal.tasks;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.Internal;
+import org.quiltmc.enigma.api.EnigmaProfile;
+import org.quiltmc.enigma.api.service.JarIndexerService;
+
+import java.util.Collection;
+import java.util.stream.Stream;
+
+import static org.quiltmc.enigma_plugin.Arguments.SIMPLE_TYPE_FIELD_NAMES_PATH;
+
+public abstract class EnigmaProfileConsumingTask extends DefaultMappingsTask {
+    @Internal(
+        "An EnigmaProfile cannot be fingerprinted. " +
+            "Up-to-date-ness is ensured by getSimpleTypeFieldNamesFiles and its source, " +
+            "MappingsExtension::getEnigmaProfileFile."
+    )
+    public abstract Property<EnigmaProfile> getEnigmaProfile();
```

Comment:
This is enough for build cache, but won't get you to config cache -- _all_ task properties need to be serialize-able or transient (in which case they may not exist at task runtime), `@Internal` or `@Input`s or `@Output`s or whatever. Question for your consideration: is there ever more than a single `EnigmaProfile` created by this plugin, at all? If the answer is no, use a build service!